Cargo.toml

```diff
@@ -10,7 +10,7 @@ keywords = ["game", "engine", "gamedev", "graphics", "bevy"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/bevyengine/bevy"
 documentation = "https://docs.rs/bevy"
-rust-version = "1.88.0"
+rust-version = "1.89.0"
```

Comment:
I'm not sure if it makes sense to bump the MSRV just for this small change.

Reply:
I personally don't see the harm: `without_provenance()` is much simpler than what we had before, and it gives Miri extra insight into what we're doing with the pointer. Bevy's MSRV policy has made it clear that we'll take advantage of newly-stabilized features if they're useful to us, so I feel it's fair game.